### PR TITLE
8253424: Add support for running pre-submit testing using GitHub Actions

### PR DESCRIPTION
**Pull request by RealCLanger**
This is the backport of the GitHub Actions workflow to JDK11 Updates

It is close to the current state of the workflow in head openjdk/jdk, except for the following modifications:

- No Windows Aarch64
- No MacOs Aarch64
- Used jtreg5.1-b01 vs. jtreg-6+1 (which means a slight modification for the steps that build it). This can/should probably be updated later
- Use Adopt OpenJDK 11.0.11 as boot JDK (I don't know whether Adoptium 11.0.12 is ready. I guess this can be changed later on as well)
- Computation of JDK_VERSION used in bundle names: It is different because 11u is an update release (e.g. we have to account for 11.0.13 vs. only 11)
- Added a step "Fix jtreg permissions" because configure on Linux/Mac would bail out otherwise. I don't know why this is not needed in head. Maybe configure was improved?
- Use gcc 9 vs. gcc 10 on Linux. I chose to take what we use in SAP's nightly tests for the moment. We could probably spend some effort later to update the code to be able to compile on gcc 10.
- No gtest support
- Make target for test runs is "run-test-prebuilt" vs. "test-prebuilt". There was some refactoring between 11 and 18.
- MacOS bundle names contain 'osx' (vs 'macos' in head)

To make the pipeline work and results look greenish, I proposed 6 other backports to be done before merging this: #180, #181, #182, #185, #186 and #187.

There are probably a few more testfixes needed to further reduce testing noise.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issues
 * [JDK-8253424](https://bugs.openjdk.java.net/browse/JDK-8253424): Add support for running pre-submit testing using GitHub Actions
 * [JDK-8253865](https://bugs.openjdk.java.net/browse/JDK-8253865): Pre-submit testing using GitHub Actions does not detect failures reliably
 * [JDK-8254054](https://bugs.openjdk.java.net/browse/JDK-8254054): Pre-submit testing using GitHub Actions should not use the deprecated set-env command
 * [JDK-8254173](https://bugs.openjdk.java.net/browse/JDK-8254173): Add Zero, Minimal hotspot targets to submit workflow
 * [JDK-8254175](https://bugs.openjdk.java.net/browse/JDK-8254175): Build no-pch configuration in debug mode for submit checks
 * [JDK-8254282](https://bugs.openjdk.java.net/browse/JDK-8254282): Add Linux x86_32 builds to submit workflow
 * [JDK-8255373](https://bugs.openjdk.java.net/browse/JDK-8255373): Submit workflow artifact name is always "test-results_.zip"
 * [JDK-8255895](https://bugs.openjdk.java.net/browse/JDK-8255895): Submit workflow artifacts miss hs_errs/replays due to ZIP include mismatch
 * [JDK-8256127](https://bugs.openjdk.java.net/browse/JDK-8256127): Add cross-compiled foreign architectures builds to submit workflow
 * [JDK-8256277](https://bugs.openjdk.java.net/browse/JDK-8256277): Github Action build on macOS should define OS and Xcode versions
 * [JDK-8256354](https://bugs.openjdk.java.net/browse/JDK-8256354): Github Action build on Windows should define OS and MSVC versions
 * [JDK-8256414](https://bugs.openjdk.java.net/browse/JDK-8256414): add optimized build to submit workflow
 * [JDK-8256393](https://bugs.openjdk.java.net/browse/JDK-8256393): Github Actions build on Linux should define OS and GCC versions
 * [JDK-8256747](https://bugs.openjdk.java.net/browse/JDK-8256747): GitHub Actions: decouple the hotspot build-only jobs from Linux x64 testing
 * [JDK-8257056](https://bugs.openjdk.java.net/browse/JDK-8257056): Submit workflow should apt-get update to avoid package installation errors
 * [JDK-8259679](https://bugs.openjdk.java.net/browse/JDK-8259679): GitHub actions should use MSVC 14.28
 * [JDK-8259924](https://bugs.openjdk.java.net/browse/JDK-8259924): GitHub actions fail on Linux x86_32 with "Could not configure libc6:i386"
 * [JDK-8260460](https://bugs.openjdk.java.net/browse/JDK-8260460): GitHub actions still fail on Linux x86_32 with "Could not configure libc6:i386"
 * [JDK-8263667](https://bugs.openjdk.java.net/browse/JDK-8263667): Avoid running GitHub actions on branches named pr/*
 * [JDK-8255305](https://bugs.openjdk.java.net/browse/JDK-8255305): Add Linux x86_32 tier1 to submit workflow
 * [JDK-8255352](https://bugs.openjdk.java.net/browse/JDK-8255352): Archive important test outputs in submit workflow

### Reviewers
 * [Severin Gehwolf](https://openjdk.java.net/census#sgehwolf) (@jerboaa - **Reviewer**) ⚠️ Review applies to 73229c2d77e709bc399ca8134d60d00bc0e148e1
 * [Aleksey Shipilev](https://openjdk.java.net/census#shade) (@shipilev - **Reviewer**) ⚠️ Review applies to 73229c2d77e709bc399ca8134d60d00bc0e148e1

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk11u-dev pull/189/head:pull/189` \
`$ git checkout pull/189`

Update a local copy of the PR: \
`$ git checkout pull/189` \
`$ git pull https://git.openjdk.java.net/jdk11u-dev pull/189/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 189`

View PR using the GUI difftool: \
`$ git pr show -t 189`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk11u-dev/pull/189.diff">https://git.openjdk.java.net/jdk11u-dev/pull/189.diff</a>

</details>

### CODE REVIEW

**Comment by shipilev** [comment]
Thank you for doing this! So we would also need to do followup additions (as separate backports), right?

8254282: Add Linux x86_32 builds to submit workflow
8254175: Build no-pch configuration in debug mode for submit checks
8254173: Add Zero, Minimal hotspot targets to submit workflow
8256414: add optimized build to submit workflow

**Comment by shipilev** [comment]
Also, from the looks of it, "8256127: Add cross-compiled foreign architectures builds to submit workflow" is also included here.

**Comment by RealCLanger** [comment]
> Thank you for doing this! So we would also need to do followup additions (as separate backports), right?
> 
> 8254282: Add Linux x86_32 builds to submit workflow
> 8254175: Build no-pch configuration in debug mode for submit checks
> 8254173: Add Zero, Minimal hotspot targets to submit workflow
> 8256414: add optimized build to submit workflow

Yes, the state of the workflow is very close to what we have in head openjdk/jdk. I'll update the description with some details and mark the important JBS items that I find as backported. Stay tuned...

File: make/conf/test-dependencies [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+# Versions and download locations for dependencies used by pre-submit testing.
+
+BOOT_JDK_VERSION=11
+JTREG_VERSION=5.1
+JTREG_BUILD=b01
+GTEST_VERSION=1.8.1
+
+LINUX_X64_BOOT_JDK_FILENAME=openjdk-11_linux-x64_bin.tar.gz
+LINUX_X64_BOOT_JDK_URL=https://github.com/AdoptOpenJDK/openjdk11-binaries/releases/download/jdk-11.0.11%2B9/OpenJDK11U-jdk_x64_linux_hotspot_11.0.11_9.tar.gz
```
Comment (jerboaa) [inline make/conf/test-dependencies:34]: Could we use Eclipse Temurin binaries now, please? Latest is:
https://github.com/adoptium/temurin11-binaries/releases/download/jdk-11.0.12%2B7/OpenJDK11U-jdk_x64_linux_hotspot_11.0.12_7.tar.gz

See:
https://github.com/adoptium/temurin11-binaries/releases/tag/jdk-11.0.12%2B7

**Review by jerboaa — COMMENTED** [review]
My preference would be to use Temurin binaries.

File: make/conf/test-dependencies (hunk 1 above)
Comment (RealCLanger) [reply make/conf/test-dependencies]: OK, I updated this. Let's see how it goes: https://github.com/RealCLanger/jdk11u-dev/actions/runs/1099309576

**Review by shipilev — APPROVED** [review]
This looks fine to me. I think these fixes are included too, but not linked to this PR:

8255305: Add Linux x86_32 tier1 to submit workflow
8255352: Archive important test outputs in submit workflow

**Comment by RealCLanger** [comment]
> This looks fine to me. I think these fixes are included too, but not linked to this PR:
> 
> 8255305: Add Linux x86_32 tier1 to submit workflow
> 8255352: Archive important test outputs in submit workflow

Thanks, I added those as well.

**Review by jerboaa — APPROVED** [review]
I cannot comment on the workflow part (know too little about them), but the test-deps look good. Thumbs up!

**Comment by RealCLanger** [comment]
> I cannot comment on the workflow part (know too little about them), but the test-deps look good. Thumbs up!

Thank you. I've flagged the bug for approval now.

**Pull request closed without merging**